sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -455,6 +455,25 @@ class ResolveSessionCatalog(
       ShowTablePropertiesCommand(
         tableName.asTableIdentifier,
         propertyKey)
+
+    case ShowFunctionsStatement(scope, pattern, functionName) =>
+      import ShowFunctionsStatement._
+      val userScope = scope.map(s => s == ALL || s == USER).getOrElse(true)
+      val systemScope = scope.map(s => s == ALL || s == SYSTEM).getOrElse(true)
+      val (db, function) = functionName match {
```

Comment:
the function name may contain catalog name, we should resolve it first.  We should match `ShowFunctionsStatement(..., SessionCatalog(_, functionName))`

Reply:
Thanks, I was a little confused with other pull request

Reply:
Looks like this is not addressed as suggested by @cloud-fan?

Reply:
I am in doubt with this in the case that the catalog is not a session catalog. We have to resolve catagog, but if the catalog is not a session catalog I think we have to show a good error message. When we do the same in other commands we do something like this to generate an exception
` if (!isSessionCatalog(catalog)) {
      throw new AnalysisException(s"$sql is only supported with v1 tables.")
    }`
If we do something like this
`case ShowFunctionsStatement(..., SessionCatalog(_, functionName))`
It becomes an unresolved exception because any case match,  and that error is not so clear.

This is the reason why I do in that way but it's posible I'm missing something.

Reply:
Currently the table loop up here doesn't respect multi-catalog. e.g. what happens for `SHOW FUNCTIONS myCata.ns1`?

You have a point that we should fail nicely for v2 catalogs. How about
```
case ShowFunctionsStatement(userScope, systemScope, pattern,
    CatalogAndIdentifierParts(catalog, functionName)) =>
  if (isSessionCatalog(catalog)) ... else fail...
```

Reply:
It can't be exactly in that way because functionName is an option. What do you think if we do it as the last changes I have pushed?